comp110-worksheet-6/OxoBoard.cs

```diff
@@ -6,53 +6,272 @@
+                            {
+                                checkRight = false;
+                            }
+
+                            if (symbolCount == itemsInARow)
+                            {
+                                winCondition = true;
+                            }
+                        }
+                        else
+                        {
+                            checkRight = false;
+                        }
+
+                    } while (checkRight && !winCondition);
+                }
+            }
+            
+
+            // If that was a win we don't need to check down and left diagonals so return the win.
+            if (winCondition)
+            {
+                return winCondition;
+            }
+
+            // Otherwise we check the down and left diagonals.
+            testX = x;
+            testY = y;
+
+            testX--;
+            testY++;
+
+            if (testX > 0 && testY < board.GetLength(1))
+            {
+                if (board[testX, testY] == targetMark)
+                {
+                    int symbolCount = 2;
+                    do
+                    {
+                        testX--;
+                        testY++;
+                        if (testX >= 0 && testY < board.GetLength(1))
+                        {
+                            if (board[testX, testY] == targetMark)
+                            {
+                                symbolCount++;
+                            } else
+                            {
+                                checkLeft = false;
+                            }
+
+                            if (symbolCount == itemsInARow)
+                            {
+                                winCondition = true;
+                            }
+                        }
+                        else
+                        {
+                            checkLeft = false;
+                        }
+
+                    } while (checkLeft && !winCondition);
+                }
+            }               
+
+            return winCondition;
+        }
+
+		// If a player has x in a row, return Mark.O or Mark.X depending on which player.
 		// Otherwise, return Mark.None.
 		public Mark GetWinner()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            int playerOCount;
+			int playerXCount;
+
+			// Checks the board vertically for x in a row
+			for (int x = 0; x < board.GetLength(0); x++)
+			{
+                playerXCount = 0;
+                playerOCount = 0;
+                for (int y = 0; y < board.GetLength(1); y++)
+				{
+					if (board[x, y] == Mark.O)
+					{
+						playerOCount++;
+					}
+					else if (board[x, y] == Mark.X)
+					{
+						playerXCount++;
+					}
+
+					if (playerOCount == itemsInARow)
+					{
+						return Mark.O;
+					} 
+					else if (playerXCount == itemsInARow)
+					{
+						return Mark.X;
+					}
+				}
```

Comment:
You don't seem to be checking if the items are consecutive -- if the height of the board is greater than `itemsInARow`, a player could win just by putting `itemsInARow` marks in a given column, not necessarily in a line?